xarray/core/dataarray.py

```diff
@@ -875,8 +875,13 @@ def encoding(self) -> dict[Any, Any]:
 
     @encoding.setter
     def encoding(self, value: Mapping[Any, Any]) -> None:
+        # deprecated (variable setter will warn)
         self.variable.encoding = dict(value)
 
+    def _set_encoding_internal(self, value: Mapping[Any, Any]) -> None:
+        """temporary method to set encoding without issuing a FutureWarning"""
+        self.variable._set_encoding_internal(value)
```

Comment:
In the future when `@encoding.setter` is gone, we will likely still wan/need a way to insert encoding attributes onto objects (coding and backends currently require this). I'm open to new names but this method is intended to be that code path.